lib/shopify_cli/theme/dev_server/hot-reload.js

```diff
@@ -1,93 +1,131 @@
 (() => {
-  function connect() {
-    const eventSource = new EventSource('/hot-reload');
-
-    eventSource.onmessage = handleUpdate;
-
-    eventSource.onopen = () => console.log('[HotReload] SSE connected.');
-
-    eventSource.onclose = () => {
-      console.log('[HotReload] SSE closed. Attempting to reconnect...');
-
-      setTimeout(connect, 5000);
+  class SectionReloadEvent extends Event {
+    constructor(filename, name, id) {
+      super('shopify:hot:section', { bubbles: true, cancelable: false });
```

Comment:
Following the [convention of the other events](https://shopify.dev/themes/architecture/sections/integrate-sections-with-the-theme-editor#section-and-block-javascript-events) we have, I believe we could rename it to `shopify: section:hotreloaded`.